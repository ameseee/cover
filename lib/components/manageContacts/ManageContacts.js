import React, { Component } from 'react';

class ManageContacts extends Component {
  constructor() {
    super();
    this.state = {
      contactName: '',
      contactNumber: '',
    };
  }

  handleChange(key, event) {
    this.setState({ [key]: event.target.value });
    console.log('handle submit');
  }

  handleSubmit= () => {
    console.log('handleClick');
    // add contact to db, re-render so they show up on a card.
  }

  handleEdit = () => {
    console.log('we are in handle Edit');
    // open up a form so they can edit.
  }

  handleRemove = () => {
    console.log('we are in handle remove');
    // delete contact from db, re-render so their card is removed
  }

  createContactCards() {
    // map over contacts and for each return:
    return (
      <article className="maange-contact-card">
        <h4>Name</h4>
        <p>Number</p>
        <button onClick={this.handleEdit}>Edit</button>
        <button onClick={this.handleRemove}>Remove</button>
      </article>
    );
  }

  render() {
    return (
      <div>
        <section className="current-contacts">
            {this.createContactCards}
        </section>

        <section className="create-contacts">
          <form>
            <input
              type="text"
              placeholder="Contact Name"
              onChange={this.handleChange}></input>
            <input
              type="text"
              placeholder="Phone Number"
              onChange={this.handleChange}></input>
            <input
              type="submit"
              onCLick={this.handleSubmit}>Save Contact</input>
          </form>
        </section>
      </div>
    );
  }
}

export default ManageContacts;
