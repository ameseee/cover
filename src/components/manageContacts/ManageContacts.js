import React, { Component } from 'react';
import firebase from '../../firebase';

class ManageContacts extends Component {
  constructor() {
    super();
    this.state = {
      contactName: '',
      contactNumber: '',
      contacts: []
    };

    this.handleChange = this.handleChange.bind(this);
    this.handleSubmit = this.handleSubmit.bind(this);
  }

  componentDidMount() {
    const itemsRef = firebase.database().ref('contacts');
    itemsRef.on('value', (snapshot) => {
      let contacts = snapshot.val();
      let newState = [];
      for (let contact in contacts) {
        newState.push({
          id: contact,
          contactName: contacts[contact].contactName,
          contactNumber: contacts[contact].contactNumber
        });
      }
      this.setState({
        contacts: newState
      });
    });
  }

  handleChange(event) {
  this.setState({
    [event.target.name]: event.target.value
  });
}

  handleSubmit(e) {
    e.preventDefault();
    const itemsRef = firebase.database().ref('contacts');
    const item = {
      contactName: this.state.contactName,
      contactNumber: this.state.contactNumber
    }
    itemsRef.push(item);
    this.setState({
      contactName: '',
      contactNumber: ''
    });
  }

  handleEdit() {
    console.log('we are in handle Edit');
    // open up a form so they can edit.
  }

  handleRemove(contact) {
    console.log('we are in handle remove', contact.id);
    const itemRef = firebase.database().ref(`/contacts/${contact.id}`);
    itemRef.remove();
    // delete contact from db, re-render so their card is removed
  }

  createContactCards() {
    // map over contacts and for each return:
    return (
      <article className="manage-contact-card">
        {/* <h4>{contact.name}</h4>
        <p>{contact.number}</p> */}
        <button onClick={this.handleEdit}>Edit</button>
        <button onClick={this.handleRemove}>Remove</button>
      </article>
    );
  }

  render() {
    return (
      <div className="manage-contacts-section">
        <h3 className="title">Manage Contacts</h3>
        <section className="current-contacts">
            <article className="existing-contact-card">
              {this.state.contacts.map(contact => {
                return (
                  <article>
                  <div>
                    <h4 className="existing-contact-name">{contact.contactName}</h4>
                    <p className="existing-contact-number">{contact.contactNumber}</p>
                  </div>
                  <button
                    className="edit-contact-btn"
                    onClick={this.handleEdit}>Edit</button>
                    <button
                      className="remove-contact-btn" onClick={() => this.handleRemove(contact)}>Remove</button>
                  </article>
                )
              })}
            </article>
        </section>

        <section className="create-contact-form">
          <h3 className="title">Add Contacts</h3>
          <form>
            <input
              className="new-contact-name"
              name="contactName"
              value={this.state.contactName}
              type="text"
              placeholder="Contact Name"
              onChange={this.handleChange}></input>
            <input
              className="new-contact-number"
              type="text"
              name="contactNumber"
              value={this.state.contactNumber}
              placeholder="Phone Number"
              onChange={this.handleChange}></input>
            <buttom
              className="save-new-contact"
              onClick={this.handleSubmit}>Save Contact</buttom>
          </form>
        </section>
      </div>
    );
  }
}

export default ManageContacts;
