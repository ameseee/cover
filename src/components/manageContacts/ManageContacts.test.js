import React from 'react';
import ManageContacts from './ManageContacts';
import { shallow } from 'enzyme';
import testSetup from '../../../__mock__/testSetup';

describe('<ManageContacts />', () => {
  let wrapper;
  let mockFn = jest.fn();

  beforeEach( () => {
    wrapper = shallow(
      <ManageContacts
        loadedContacts={[
          { contactName: 'Amy',
            contactNumber: '5756441355'},
          { contactName: 'Cindy',
            contactNumber: '3035551234'},
        ]}/>);
  });

  it('should exist', () => {
    expect(wrapper).toBeDefined();
  });

  it('should have an empty default state of contactName', () => {
    expect(wrapper.state().contactName).toEqual('');
  });

  it('should have an empty default state of contactNumber', () => {
    expect(wrapper.state().contactNumber).toEqual('');
  });

  it('should render a title', () => {
    const title = wrapper.find('.section-title');

    expect(title.length).toEqual(1);
  });

  it('should render all contacts', () => {
    const allContacts = wrapper.find('.all-contacts');

    expect(allContacts.length).toEqual(1);
  });

  //figure out how to give props to the component, give 2 fake contacts

  it('should render two contact cards', () => {
    const cards = wrapper.find('.existing-contact-card');

    expect(cards.length).toEqual(2);
  });

  it('should render two contact names', () => {
    const names = wrapper.find('.existing-contact-name');

    expect(names.length).toEqual(2);
  });

  it('should render two contact numbers', () => {
    const numbers = wrapper.find('.existing-contact-name');

    expect(numbers.length).toEqual(2);
  });

  it('should render two remove buttons', () => {
    const removeBtn = wrapper.find('.remove-contact-btn');

    expect(removeBtn.length).toEqual(2);
  });

  it('should run a function to remove a contact', () => {
    const removeBtn = wrapper.find('.remove-contact-btn').first();

    wrapper.instance().handleRemove = mockFn;
    removeBtn.simulate('click');
    expect(mockFn).toHaveBeenCalledTimes(1);
  });

  it('should render a section to add new contacts', () => {
    const formSection = wrapper.find('.create-contact-form');

    expect(formSection.length).toEqual(1);
  });

  it('should render a title for the form', () => {
    const add = wrapper.find('.add');

    expect(add.length).toEqual(1);
  });

  it('should render two inputs', () => {
    const input = wrapper.find('input');

    expect(input.length).toEqual(2);
  });

//not passing!!! it says state is still empty
  it('should change state on input', () => {
    const nameInput = wrapper.find('.new-contact-name');
    const numberInput = wrapper.find('.new-contact-number');

    nameInput.simulate('change', { target: { value: 'Amy' } });
    numberInput.simulate('change', { target: { value: '5756441355' } });
    expect(wrapper.state().contactName).toEqual('Amy');
    expect(wrapper.state().contactNumber).toEqual('5756441355');
  });

  it('should render a button to save new contact', () => {
    const saveBtn = wrapper.find('.save-new-contact');

    expect(saveBtn.length).toEqual(1);
  });

//not passing - mad about firebase shit
  it('should run a function to save new contact', () => {
    const saveBtn = wrapper.find('.save-new-contact');

    //wrapper.instance().handleSubmit = mockFn;
    saveBtn.simulate('click', { preventDefault() {} });
    expect(mockFn).toHaveBeenCalledTimes(1);
  });

//not passing - mad about firebase shit
  it('should change state when a new user is added', () => {
    const nameInput = wrapper.find('.new-contact-name');
    const numberInput = wrapper.find('.new-contact-number');
    const saveBtn = wrapper.find('.save-new-contact');

    //wrapper.instance().handleSubmit = mockFn;
    saveBtn.simulate('click', { preventDefault() {} });
    expect(wrapper.state().contactName).toEqual('');
    expect(wrapper.state().contactNumber).toEqual('');
  });

  it('should match snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });

});
