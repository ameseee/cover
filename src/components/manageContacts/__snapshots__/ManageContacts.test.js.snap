// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ManageContacts /> should match snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ManageContacts
    loadedContacts={
      Array [
        Object {
          "contactName": "Amy",
          "contactNumber": "5756441355",
        },
        Object {
          "contactName": "Cindy",
          "contactNumber": "3035551234",
        },
      ]
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": null,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <h3
          className="section-title"
        >
          Manage Contacts
        </h3>,
        <section
          className="current-contacts"
        >
          <article
            className="all-contacts"
          >
            <article
              className="existing-contact-card"
            >
              <div>
                <h4
                  className="existing-contact-name"
                >
                  Amy
                </h4>
                <p
                  className="existing-contact-number"
                >
                  5756441355
                </p>
              </div>
              <button
                className="edit-contact-btn"
                onClick={[Function]}
              >
                Edit
              </button>
              <button
                className="remove-contact-btn"
                onClick={[Function]}
              >
                Remove
              </button>
            </article>
            <article
              className="existing-contact-card"
            >
              <div>
                <h4
                  className="existing-contact-name"
                >
                  Cindy
                </h4>
                <p
                  className="existing-contact-number"
                >
                  3035551234
                </p>
              </div>
              <button
                className="edit-contact-btn"
                onClick={[Function]}
              >
                Edit
              </button>
              <button
                className="remove-contact-btn"
                onClick={[Function]}
              >
                Remove
              </button>
            </article>
          </article>
        </section>,
        <section
          className="create-contact-form"
        >
          <h3
            className="title add"
          >
            Add Contacts
          </h3>
          <form>
            <input
              className="new-contact-name"
              name="contactName"
              onChange={[Function]}
              placeholder="Contact Name"
              type="text"
              value=""
            />
            <input
              className="new-contact-number"
              name="contactNumber"
              onChange={[Function]}
              placeholder="Phone Number"
              type="text"
              value=""
            />
            <button
              className="save-new-contact"
              onClick={[Function]}
            >
              Save Contact
            </button>
          </form>
        </section>,
      ],
      "className": "manage-contacts-section",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": null,
        "nodeType": "host",
        "props": Object {
          "children": "Manage Contacts",
          "className": "section-title",
        },
        "ref": null,
        "rendered": "Manage Contacts",
        "type": "h3",
      },
      Object {
        "instance": null,
        "key": null,
        "nodeType": "host",
        "props": Object {
          "children": <article
            className="all-contacts"
          >
            <article
              className="existing-contact-card"
            >
              <div>
                <h4
                  className="existing-contact-name"
                >
                  Amy
                </h4>
                <p
                  className="existing-contact-number"
                >
                  5756441355
                </p>
              </div>
              <button
                className="edit-contact-btn"
                onClick={[Function]}
              >
                Edit
              </button>
              <button
                className="remove-contact-btn"
                onClick={[Function]}
              >
                Remove
              </button>
            </article>
            <article
              className="existing-contact-card"
            >
              <div>
                <h4
                  className="existing-contact-name"
                >
                  Cindy
                </h4>
                <p
                  className="existing-contact-number"
                >
                  3035551234
                </p>
              </div>
              <button
                className="edit-contact-btn"
                onClick={[Function]}
              >
                Edit
              </button>
              <button
                className="remove-contact-btn"
                onClick={[Function]}
              >
                Remove
              </button>
            </article>
          </article>,
          "className": "current-contacts",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": null,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <article
                className="existing-contact-card"
              >
                <div>
                  <h4
                    className="existing-contact-name"
                  >
                    Amy
                  </h4>
                  <p
                    className="existing-contact-number"
                  >
                    5756441355
                  </p>
                </div>
                <button
                  className="edit-contact-btn"
                  onClick={[Function]}
                >
                  Edit
                </button>
                <button
                  className="remove-contact-btn"
                  onClick={[Function]}
                >
                  Remove
                </button>
              </article>,
              <article
                className="existing-contact-card"
              >
                <div>
                  <h4
                    className="existing-contact-name"
                  >
                    Cindy
                  </h4>
                  <p
                    className="existing-contact-number"
                  >
                    3035551234
                  </p>
                </div>
                <button
                  className="edit-contact-btn"
                  onClick={[Function]}
                >
                  Edit
                </button>
                <button
                  className="remove-contact-btn"
                  onClick={[Function]}
                >
                  Remove
                </button>
              </article>,
            ],
            "className": "all-contacts",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": "5756441355",
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <div>
                    <h4
                      className="existing-contact-name"
                    >
                      Amy
                    </h4>
                    <p
                      className="existing-contact-number"
                    >
                      5756441355
                    </p>
                  </div>,
                  <button
                    className="edit-contact-btn"
                    onClick={[Function]}
                  >
                    Edit
                  </button>,
                  <button
                    className="remove-contact-btn"
                    onClick={[Function]}
                  >
                    Remove
                  </button>,
                ],
                "className": "existing-contact-card",
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": null,
                  "nodeType": "host",
                  "props": Object {
                    "children": Array [
                      <h4
                        className="existing-contact-name"
                      >
                        Amy
                      </h4>,
                      <p
                        className="existing-contact-number"
                      >
                        5756441355
                      </p>,
                    ],
                  },
                  "ref": null,
                  "rendered": Array [
                    Object {
                      "instance": null,
                      "key": null,
                      "nodeType": "host",
                      "props": Object {
                        "children": "Amy",
                        "className": "existing-contact-name",
                      },
                      "ref": null,
                      "rendered": "Amy",
                      "type": "h4",
                    },
                    Object {
                      "instance": null,
                      "key": null,
                      "nodeType": "host",
                      "props": Object {
                        "children": "5756441355",
                        "className": "existing-contact-number",
                      },
                      "ref": null,
                      "rendered": "5756441355",
                      "type": "p",
                    },
                  ],
                  "type": "div",
                },
                Object {
                  "instance": null,
                  "key": null,
                  "nodeType": "host",
                  "props": Object {
                    "children": "Edit",
                    "className": "edit-contact-btn",
                    "onClick": [Function],
                  },
                  "ref": null,
                  "rendered": "Edit",
                  "type": "button",
                },
                Object {
                  "instance": null,
                  "key": null,
                  "nodeType": "host",
                  "props": Object {
                    "children": "Remove",
                    "className": "remove-contact-btn",
                    "onClick": [Function],
                  },
                  "ref": null,
                  "rendered": "Remove",
                  "type": "button",
                },
              ],
              "type": "article",
            },
            Object {
              "instance": null,
              "key": "3035551234",
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <div>
                    <h4
                      className="existing-contact-name"
                    >
                      Cindy
                    </h4>
                    <p
                      className="existing-contact-number"
                    >
                      3035551234
                    </p>
                  </div>,
                  <button
                    className="edit-contact-btn"
                    onClick={[Function]}
                  >
                    Edit
                  </button>,
                  <button
                    className="remove-contact-btn"
                    onClick={[Function]}
                  >
                    Remove
                  </button>,
                ],
                "className": "existing-contact-card",
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": null,
                  "nodeType": "host",
                  "props": Object {
                    "children": Array [
                      <h4
                        className="existing-contact-name"
                      >
                        Cindy
                      </h4>,
                      <p
                        className="existing-contact-number"
                      >
                        3035551234
                      </p>,
                    ],
                  },
                  "ref": null,
                  "rendered": Array [
                    Object {
                      "instance": null,
                      "key": null,
                      "nodeType": "host",
                      "props": Object {
                        "children": "Cindy",
                        "className": "existing-contact-name",
                      },
                      "ref": null,
                      "rendered": "Cindy",
                      "type": "h4",
                    },
                    Object {
                      "instance": null,
                      "key": null,
                      "nodeType": "host",
                      "props": Object {
                        "children": "3035551234",
                        "className": "existing-contact-number",
                      },
                      "ref": null,
                      "rendered": "3035551234",
                      "type": "p",
                    },
                  ],
                  "type": "div",
                },
                Object {
                  "instance": null,
                  "key": null,
                  "nodeType": "host",
                  "props": Object {
                    "children": "Edit",
                    "className": "edit-contact-btn",
                    "onClick": [Function],
                  },
                  "ref": null,
                  "rendered": "Edit",
                  "type": "button",
                },
                Object {
                  "instance": null,
                  "key": null,
                  "nodeType": "host",
                  "props": Object {
                    "children": "Remove",
                    "className": "remove-contact-btn",
                    "onClick": [Function],
                  },
                  "ref": null,
                  "rendered": "Remove",
                  "type": "button",
                },
              ],
              "type": "article",
            },
          ],
          "type": "article",
        },
        "type": "section",
      },
      Object {
        "instance": null,
        "key": null,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <h3
              className="title add"
            >
              Add Contacts
            </h3>,
            <form>
              <input
                className="new-contact-name"
                name="contactName"
                onChange={[Function]}
                placeholder="Contact Name"
                type="text"
                value=""
              />
              <input
                className="new-contact-number"
                name="contactNumber"
                onChange={[Function]}
                placeholder="Phone Number"
                type="text"
                value=""
              />
              <button
                className="save-new-contact"
                onClick={[Function]}
              >
                Save Contact
              </button>
            </form>,
          ],
          "className": "create-contact-form",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": null,
            "nodeType": "host",
            "props": Object {
              "children": "Add Contacts",
              "className": "title add",
            },
            "ref": null,
            "rendered": "Add Contacts",
            "type": "h3",
          },
          Object {
            "instance": null,
            "key": null,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <input
                  className="new-contact-name"
                  name="contactName"
                  onChange={[Function]}
                  placeholder="Contact Name"
                  type="text"
                  value=""
                />,
                <input
                  className="new-contact-number"
                  name="contactNumber"
                  onChange={[Function]}
                  placeholder="Phone Number"
                  type="text"
                  value=""
                />,
                <button
                  className="save-new-contact"
                  onClick={[Function]}
                >
                  Save Contact
                </button>,
              ],
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": null,
                "nodeType": "host",
                "props": Object {
                  "className": "new-contact-name",
                  "name": "contactName",
                  "onChange": [Function],
                  "placeholder": "Contact Name",
                  "type": "text",
                  "value": "",
                },
                "ref": null,
                "rendered": null,
                "type": "input",
              },
              Object {
                "instance": null,
                "key": null,
                "nodeType": "host",
                "props": Object {
                  "className": "new-contact-number",
                  "name": "contactNumber",
                  "onChange": [Function],
                  "placeholder": "Phone Number",
                  "type": "text",
                  "value": "",
                },
                "ref": null,
                "rendered": null,
                "type": "input",
              },
              Object {
                "instance": null,
                "key": null,
                "nodeType": "host",
                "props": Object {
                  "children": "Save Contact",
                  "className": "save-new-contact",
                  "onClick": [Function],
                },
                "ref": null,
                "rendered": "Save Contact",
                "type": "button",
              },
            ],
            "type": "form",
          },
        ],
        "type": "section",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": null,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h3
            className="section-title"
          >
            Manage Contacts
          </h3>,
          <section
            className="current-contacts"
          >
            <article
              className="all-contacts"
            >
              <article
                className="existing-contact-card"
              >
                <div>
                  <h4
                    className="existing-contact-name"
                  >
                    Amy
                  </h4>
                  <p
                    className="existing-contact-number"
                  >
                    5756441355
                  </p>
                </div>
                <button
                  className="edit-contact-btn"
                  onClick={[Function]}
                >
                  Edit
                </button>
                <button
                  className="remove-contact-btn"
                  onClick={[Function]}
                >
                  Remove
                </button>
              </article>
              <article
                className="existing-contact-card"
              >
                <div>
                  <h4
                    className="existing-contact-name"
                  >
                    Cindy
                  </h4>
                  <p
                    className="existing-contact-number"
                  >
                    3035551234
                  </p>
                </div>
                <button
                  className="edit-contact-btn"
                  onClick={[Function]}
                >
                  Edit
                </button>
                <button
                  className="remove-contact-btn"
                  onClick={[Function]}
                >
                  Remove
                </button>
              </article>
            </article>
          </section>,
          <section
            className="create-contact-form"
          >
            <h3
              className="title add"
            >
              Add Contacts
            </h3>
            <form>
              <input
                className="new-contact-name"
                name="contactName"
                onChange={[Function]}
                placeholder="Contact Name"
                type="text"
                value=""
              />
              <input
                className="new-contact-number"
                name="contactNumber"
                onChange={[Function]}
                placeholder="Phone Number"
                type="text"
                value=""
              />
              <button
                className="save-new-contact"
                onClick={[Function]}
              >
                Save Contact
              </button>
            </form>
          </section>,
        ],
        "className": "manage-contacts-section",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": null,
          "nodeType": "host",
          "props": Object {
            "children": "Manage Contacts",
            "className": "section-title",
          },
          "ref": null,
          "rendered": "Manage Contacts",
          "type": "h3",
        },
        Object {
          "instance": null,
          "key": null,
          "nodeType": "host",
          "props": Object {
            "children": <article
              className="all-contacts"
            >
              <article
                className="existing-contact-card"
              >
                <div>
                  <h4
                    className="existing-contact-name"
                  >
                    Amy
                  </h4>
                  <p
                    className="existing-contact-number"
                  >
                    5756441355
                  </p>
                </div>
                <button
                  className="edit-contact-btn"
                  onClick={[Function]}
                >
                  Edit
                </button>
                <button
                  className="remove-contact-btn"
                  onClick={[Function]}
                >
                  Remove
                </button>
              </article>
              <article
                className="existing-contact-card"
              >
                <div>
                  <h4
                    className="existing-contact-name"
                  >
                    Cindy
                  </h4>
                  <p
                    className="existing-contact-number"
                  >
                    3035551234
                  </p>
                </div>
                <button
                  className="edit-contact-btn"
                  onClick={[Function]}
                >
                  Edit
                </button>
                <button
                  className="remove-contact-btn"
                  onClick={[Function]}
                >
                  Remove
                </button>
              </article>
            </article>,
            "className": "current-contacts",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": null,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <article
                  className="existing-contact-card"
                >
                  <div>
                    <h4
                      className="existing-contact-name"
                    >
                      Amy
                    </h4>
                    <p
                      className="existing-contact-number"
                    >
                      5756441355
                    </p>
                  </div>
                  <button
                    className="edit-contact-btn"
                    onClick={[Function]}
                  >
                    Edit
                  </button>
                  <button
                    className="remove-contact-btn"
                    onClick={[Function]}
                  >
                    Remove
                  </button>
                </article>,
                <article
                  className="existing-contact-card"
                >
                  <div>
                    <h4
                      className="existing-contact-name"
                    >
                      Cindy
                    </h4>
                    <p
                      className="existing-contact-number"
                    >
                      3035551234
                    </p>
                  </div>
                  <button
                    className="edit-contact-btn"
                    onClick={[Function]}
                  >
                    Edit
                  </button>
                  <button
                    className="remove-contact-btn"
                    onClick={[Function]}
                  >
                    Remove
                  </button>
                </article>,
              ],
              "className": "all-contacts",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": "5756441355",
                "nodeType": "host",
                "props": Object {
                  "children": Array [
                    <div>
                      <h4
                        className="existing-contact-name"
                      >
                        Amy
                      </h4>
                      <p
                        className="existing-contact-number"
                      >
                        5756441355
                      </p>
                    </div>,
                    <button
                      className="edit-contact-btn"
                      onClick={[Function]}
                    >
                      Edit
                    </button>,
                    <button
                      className="remove-contact-btn"
                      onClick={[Function]}
                    >
                      Remove
                    </button>,
                  ],
                  "className": "existing-contact-card",
                },
                "ref": null,
                "rendered": Array [
                  Object {
                    "instance": null,
                    "key": null,
                    "nodeType": "host",
                    "props": Object {
                      "children": Array [
                        <h4
                          className="existing-contact-name"
                        >
                          Amy
                        </h4>,
                        <p
                          className="existing-contact-number"
                        >
                          5756441355
                        </p>,
                      ],
                    },
                    "ref": null,
                    "rendered": Array [
                      Object {
                        "instance": null,
                        "key": null,
                        "nodeType": "host",
                        "props": Object {
                          "children": "Amy",
                          "className": "existing-contact-name",
                        },
                        "ref": null,
                        "rendered": "Amy",
                        "type": "h4",
                      },
                      Object {
                        "instance": null,
                        "key": null,
                        "nodeType": "host",
                        "props": Object {
                          "children": "5756441355",
                          "className": "existing-contact-number",
                        },
                        "ref": null,
                        "rendered": "5756441355",
                        "type": "p",
                      },
                    ],
                    "type": "div",
                  },
                  Object {
                    "instance": null,
                    "key": null,
                    "nodeType": "host",
                    "props": Object {
                      "children": "Edit",
                      "className": "edit-contact-btn",
                      "onClick": [Function],
                    },
                    "ref": null,
                    "rendered": "Edit",
                    "type": "button",
                  },
                  Object {
                    "instance": null,
                    "key": null,
                    "nodeType": "host",
                    "props": Object {
                      "children": "Remove",
                      "className": "remove-contact-btn",
                      "onClick": [Function],
                    },
                    "ref": null,
                    "rendered": "Remove",
                    "type": "button",
                  },
                ],
                "type": "article",
              },
              Object {
                "instance": null,
                "key": "3035551234",
                "nodeType": "host",
                "props": Object {
                  "children": Array [
                    <div>
                      <h4
                        className="existing-contact-name"
                      >
                        Cindy
                      </h4>
                      <p
                        className="existing-contact-number"
                      >
                        3035551234
                      </p>
                    </div>,
                    <button
                      className="edit-contact-btn"
                      onClick={[Function]}
                    >
                      Edit
                    </button>,
                    <button
                      className="remove-contact-btn"
                      onClick={[Function]}
                    >
                      Remove
                    </button>,
                  ],
                  "className": "existing-contact-card",
                },
                "ref": null,
                "rendered": Array [
                  Object {
                    "instance": null,
                    "key": null,
                    "nodeType": "host",
                    "props": Object {
                      "children": Array [
                        <h4
                          className="existing-contact-name"
                        >
                          Cindy
                        </h4>,
                        <p
                          className="existing-contact-number"
                        >
                          3035551234
                        </p>,
                      ],
                    },
                    "ref": null,
                    "rendered": Array [
                      Object {
                        "instance": null,
                        "key": null,
                        "nodeType": "host",
                        "props": Object {
                          "children": "Cindy",
                          "className": "existing-contact-name",
                        },
                        "ref": null,
                        "rendered": "Cindy",
                        "type": "h4",
                      },
                      Object {
                        "instance": null,
                        "key": null,
                        "nodeType": "host",
                        "props": Object {
                          "children": "3035551234",
                          "className": "existing-contact-number",
                        },
                        "ref": null,
                        "rendered": "3035551234",
                        "type": "p",
                      },
                    ],
                    "type": "div",
                  },
                  Object {
                    "instance": null,
                    "key": null,
                    "nodeType": "host",
                    "props": Object {
                      "children": "Edit",
                      "className": "edit-contact-btn",
                      "onClick": [Function],
                    },
                    "ref": null,
                    "rendered": "Edit",
                    "type": "button",
                  },
                  Object {
                    "instance": null,
                    "key": null,
                    "nodeType": "host",
                    "props": Object {
                      "children": "Remove",
                      "className": "remove-contact-btn",
                      "onClick": [Function],
                    },
                    "ref": null,
                    "rendered": "Remove",
                    "type": "button",
                  },
                ],
                "type": "article",
              },
            ],
            "type": "article",
          },
          "type": "section",
        },
        Object {
          "instance": null,
          "key": null,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <h3
                className="title add"
              >
                Add Contacts
              </h3>,
              <form>
                <input
                  className="new-contact-name"
                  name="contactName"
                  onChange={[Function]}
                  placeholder="Contact Name"
                  type="text"
                  value=""
                />
                <input
                  className="new-contact-number"
                  name="contactNumber"
                  onChange={[Function]}
                  placeholder="Phone Number"
                  type="text"
                  value=""
                />
                <button
                  className="save-new-contact"
                  onClick={[Function]}
                >
                  Save Contact
                </button>
              </form>,
            ],
            "className": "create-contact-form",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": null,
              "nodeType": "host",
              "props": Object {
                "children": "Add Contacts",
                "className": "title add",
              },
              "ref": null,
              "rendered": "Add Contacts",
              "type": "h3",
            },
            Object {
              "instance": null,
              "key": null,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <input
                    className="new-contact-name"
                    name="contactName"
                    onChange={[Function]}
                    placeholder="Contact Name"
                    type="text"
                    value=""
                  />,
                  <input
                    className="new-contact-number"
                    name="contactNumber"
                    onChange={[Function]}
                    placeholder="Phone Number"
                    type="text"
                    value=""
                  />,
                  <button
                    className="save-new-contact"
                    onClick={[Function]}
                  >
                    Save Contact
                  </button>,
                ],
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": null,
                  "nodeType": "host",
                  "props": Object {
                    "className": "new-contact-name",
                    "name": "contactName",
                    "onChange": [Function],
                    "placeholder": "Contact Name",
                    "type": "text",
                    "value": "",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": "input",
                },
                Object {
                  "instance": null,
                  "key": null,
                  "nodeType": "host",
                  "props": Object {
                    "className": "new-contact-number",
                    "name": "contactNumber",
                    "onChange": [Function],
                    "placeholder": "Phone Number",
                    "type": "text",
                    "value": "",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": "input",
                },
                Object {
                  "instance": null,
                  "key": null,
                  "nodeType": "host",
                  "props": Object {
                    "children": "Save Contact",
                    "className": "save-new-contact",
                    "onClick": [Function],
                  },
                  "ref": null,
                  "rendered": "Save Contact",
                  "type": "button",
                },
              ],
              "type": "form",
            },
          ],
          "type": "section",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
